### PR TITLE
implemented test displaying list of flashcards

### PR DESCRIPTION
**Pull request by darionauta**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Write an integration test covering displaying flashcards

### Requirements
You need to cover following scenarios:
1. It should display flashcards in the list properly

### Code Review

#### package.json
The changes in this file are correct. The test command has been updated to use `vitest` instead of `jest`. Also, `vitest` and `jsdom` have been added as dev dependencies which are required for the new testing setup.

#### src/layouts/CardsList/__tests__/CardsList.spec.tsx
The test case added in this file correctly tests the scenario mentioned in the requirements. It renders a list of flashcards and checks if they are displayed properly. However, there is a minor issue with the test data. All the cards have the same `_id` which is not realistic and could potentially hide issues related to handling of unique ids.

#### src/test/setup.ts
This file has been correctly added to import `@testing-library/jest-dom` which provides custom jest matchers that are used in the test.

#### vite.config.ts
The changes in this file are correct. The configuration for `vitest` has been added and it includes the setup file and specifies the environment as `jsdom`.

### Summary
The code changes cover the task requirements. However, there is a minor issue with the test data in `src/layouts/CardsList/__tests__/CardsList.spec.tsx` where all cards have the same `_id`. This should be corrected to ensure the test is realistic.

File: src/layouts/CardsList/__tests__/CardsList.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    it('should contain at least one child element', () => {
+        
+        const mockCards = [
+            { _id: "1", front: "Marko", back: "Polo" },
+            { _id: "1", front: "Marko", back: "Polo" },
+            { _id: "1", front: "Marko", back: "Polo" },
+        ];
+        
+        render(
+            <CardsList>
```
Comment (Sumick) [inline src/layouts/CardsList/__tests__/CardsList.spec.tsx]: **issue:** This is not correct integration test, as you should operate on internal react state. You should mock get request here

**Review by Sumick — CHANGES_REQUESTED** [review]
This is not correct integration test, as you should operate on react state - not mapping elements like this. You need to mock GET request first. This looks more like unit test which tests rendering children elements, not whole integration between application logic, and UI

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
Write an integration test covering displaying flashcards

### Requirements
You need to cover following scenarios:
1. It should display flashcards in the list properly

### Code Review

#### package.json
The changes in this file are correct. The test script was updated to use `vitest` instead of `jest`. The `vitest` and `jsdom` packages were added to the devDependencies.

#### src/layouts/CardsList/__tests__/CardsList.spec.tsx
The test case in this file meets the task requirements. It tests if the CardsList component displays the flashcards properly. The test renders the CardsList component with a list of mock cards and checks if all cards are displayed.

#### src/test/setup.ts
This file was added correctly. It imports '@testing-library/jest-dom' which provides custom jest matchers that you can use to extend jest. These will make your tests more declarative, clear to read, and to maintain.

#### vite.config.ts
The changes in this file are correct. The test configuration for `vitest` was added. The environment was set to 'jsdom', setupFiles were set to './src/test/setup.ts', and css was enabled.

### Conclusion
The code changes cover the task requirements. All changes are correct and there are no bugs in the changed code.

File: src/layouts/CardsList/__tests__/CardsList.spec.tsx (hunk 1 above)
Comment (darionauta) [reply src/layouts/CardsList/__tests__/CardsList.spec.tsx]: Can you check my test again?

**Review by Sumick — APPROVED** [review]
Looks good now!:)

**Pull request merged**